### PR TITLE
feat(rds_instance_transport_encrypted): add new check

### LINKED ISSUE #1951: Implement a new check for RDS to require encrypted connections
**Is your feature request related to a problem? Please describe.**
The default DB parameter groups for RDS has the rds.force_ssl parameter set to 0. This parameter can be set to 1 for MSSQL and PostgreSQL to require that connections made to the database are protected by SSL.

**Describe the solution you'd like**
Implement a new check to ensure that the DB parameter group enforces SSL. Rather than just checking if the default parameter group is in use, the check should determine if the rds.force_ssl parameter is set to 1.

This check could also raise an informational risk finding advising users to check if SSL connections are required for Oracle DB, MySQL, and MariaDB. These checks need to be performed within the DB.

**Describe alternatives you've considered**
None

**Additional context**
[None](https://aws.amazon.com/premiumsupport/knowledge-center/rds-connect-ssl-connection/)

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Description

Add new check called `rds_instance_transport_encrypted`

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+from prowler.lib.check.models import Check, Check_Report_AWS
+from prowler.providers.aws.services.rds.rds_client import rds_client
+
+
+class rds_instance_transport_encrypted(Check):
+    def execute(self):
+        findings = []
+        supported_engines = ["sqlserver-ex", "postgres"]
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted.py]:
```suggestion
        supported_engines = ["sqlserver", "postgres"]
```

File: prowler/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        for db_instance in rds_client.db_instances:
+            report = Check_Report_AWS(self.metadata())
+            report.region = db_instance.region
+            report.resource_id = db_instance.id
+            report.status = "FAIL"
+            report.status_extended = (
+                f"RDS Instance {db_instance.id} connections are not encrypted."
+            )
+            # Check only RDS SQL Server or PostgreSQL engines
+            if db_instance.engine in supported_engines:
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted.py]:
```suggestion
            if any(engine in db_instance.engine for engine in supported_engines):
```

**Review by MrCloudSec — COMMENTED** [review]
Thanks to @Fennerr for the idea!

File: prowler/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted.metadata.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  "Remediation": {
+    "Code": {
+      "CLI": "aws rds modify-db-parameter-group --region <REGION_NAME> --db-parameter-group-name <PARAMETER_GROUP_NAME> --parameters ParameterName='rds.force_ssl',ParameterValue='1',ApplyMethod='pending-reboot'",
+      "NativeIaC": "",
+      "Other": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/aws/RDS/transport-encryption.html",
+      "Terraform": ""
+    },
+    "Recommendation": {
+      "Text": "Ensure that Microsoft SQL Server and PostgreSQL instances provisioned with Amazon RDS have Transport Encryption feature enabled in order to meet security and compliance requirements.",
+      "Url": "https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/PostgreSQL.Concepts.General.SSL.html"
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted.metadata.json]:
```suggestion
      "Url": "https://aws.amazon.com/premiumsupport/knowledge-center/rds-connect-ssl-connection/"
```

File: prowler/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted.metadata.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  "CheckTitle": "Check if RDS instances client connections are encrypted.",
+  "CheckType": [],
+  "ServiceName": "rds",
+  "SubServiceName": "",
+  "ResourceIdTemplate": "arn:aws:rds:region:account-id:db-instance",
+  "Severity": "high",
+  "ResourceType": "AwsRdsDbInstance",
+  "Description": "Check if RDS instances client connections are encrypted.",
+  "Risk": "If not enabled sensitive information at transit is not protected.",
+  "RelatedUrl": "https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/PostgreSQL.Concepts.General.SSL.html",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted.metadata.json]:
```suggestion
  "RelatedUrl": "https://aws.amazon.com/premiumsupport/knowledge-center/rds-connect-ssl-connection/",
```

File: prowler/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted.metadata.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    "Recommendation": {
+      "Text": "Ensure that Microsoft SQL Server and PostgreSQL instances provisioned with Amazon RDS have Transport Encryption feature enabled in order to meet security and compliance requirements.",
+      "Url": "https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/PostgreSQL.Concepts.General.SSL.html"
+    }
+  },
+  "Categories": [],
+  "Tags": {
+    "Tag1Key": "value",
+    "Tag2Key": "value"
+  },
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted.metadata.json]:
```suggestion
```

File: prowler/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted.metadata.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+      "NativeIaC": "",
+      "Other": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/aws/RDS/transport-encryption.html",
+      "Terraform": ""
+    },
+    "Recommendation": {
+      "Text": "Ensure that Microsoft SQL Server and PostgreSQL instances provisioned with Amazon RDS have Transport Encryption feature enabled in order to meet security and compliance requirements.",
+      "Url": "https://aws.amazon.com/premiumsupport/knowledge-center/rds-connect-ssl-connection/"
+    }
+  },
+  "Categories": [],
```
Comment (toniblyx) [inline prowler/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted.metadata.json]:
```suggestion
  "Categories": [ "encryption" ],
```

File: prowler/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted.metadata.json [hunk 7]
```diff
@@ -0,0 +1,30 @@
+{
+  "Provider": "aws",
+  "CheckID": "rds_instance_transport_encrypted",
+  "CheckTitle": "Check if RDS instances client connections are encrypted.",
```
Comment (toniblyx) [inline prowler/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted.metadata.json]:
```suggestion
  "CheckTitle": "Check if RDS instances client connections are encrypted (Microsoft SQL Server and PostgreSQL).",
```

**Review by toniblyx — CHANGES_REQUESTED** [review]

**Comment by Fennerr** [comment]
Just moving across the info from the slack chat I had with Sergio to here:

For Oracle databases the rds client also needs to retrieve the option groups, and then check for the SQLNET.SSL_VERSION option to that group. https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Appendix.Oracle.Options.SSL.html#Appendix.Oracle.Options.SSL.OptionGroup
The following values are allowed for this option setting:

    "1.0" - Clients can connect to the DB instance using TLS 1.0 only.
    "1.2" - Clients can connect to the DB instance using TLS 1.2 only.
    "1.2 or 1.0" - Clients can connect to the DB instance using either TLS 1.2 or 1.0.

The check can also check if it is set to use TLS 1.0
> For existing Oracle SSL options, SQLNET.SSL_VERSION is set to "1.0" automatically. You can change the setting, if necessary.

I can add this to the current check, or it can be a separate check

File: prowler/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted.metadata.json [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  "CheckType": [],
+  "ServiceName": "rds",
+  "SubServiceName": "",
+  "ResourceIdTemplate": "arn:aws:rds:region:account-id:db-instance",
+  "Severity": "high",
+  "ResourceType": "AwsRdsDbInstance",
+  "Description": "Check if RDS instances client connections are encrypted.",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted.metadata.json]:
```suggestion
  "Description": "Check if RDS instances client connections are encrypted (Microsoft SQL Server and PostgreSQL).",
```

**Comment by MrCloudSec** [comment]
> Just moving across the info from the slack chat I had with Sergio to here:
> 
> For Oracle databases the rds client also needs to retrieve the option groups, and then check for the SQLNET.SSL_VERSION option to that group. https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Appendix.Oracle.Options.SSL.html#Appendix.Oracle.Options.SSL.OptionGroup The following values are allowed for this option setting:
> 
> ```
> "1.0" - Clients can connect to the DB instance using TLS 1.0 only.
> "1.2" - Clients can connect to the DB instance using TLS 1.2 only.
> "1.2 or 1.0" - Clients can connect to the DB instance using either TLS 1.2 or 1.0.
> ```
> 
> The check can also check if it is set to use TLS 1.0
> 
> > For existing Oracle SSL options, SQLNET.SSL_VERSION is set to "1.0" automatically. You can change the setting, if necessary.
> 
> I can add this to the current check, or it can be a separate check

@Fennerr, this is a great idea, it would be awesome if you can do it, we have include it in our roadmap anyway.
If you finally do it, please create a new PR as we want to include this check in the next release that we are going to do tomorrow. Thanks!

**Review by toniblyx — APPROVED** [review]

**Pull request merged**